src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -31,26 +31,129 @@ var normalizeSectionName = function (name) {
+        if (dateString.length === 6)
+            return year + '-' + month;
+        var day = dateString.substring(6, 8);
+        if (!validDate(year, month - 1, day))
+            throw `Invalid day: ${dateString}`;
+        return year + '-' + month + '-' + day;
     }
-    var day = ds.slice(6, 8);
-    var hour = ds.slice(8, 10);
-    var minute = ds.slice(10, 12);
-    var second = ds.slice(12, 14);
-    var millisecond = ds.slice(14, 17);
-    return (new Date(Date.UTC(year, monthIndex, day, hour, minute, second, millisecond)));
+    throw `Bad input for Date type in ${dateString}`;
+};
+
+
+// handling the date format here
+var getDate = function (dateString) {
+    if (!validString(dateString))
+        return '';
+    var ds = validDigitString(dateString);
+    return convertDate(ds);
 };
 
 module.exports.internal = {
     getDate: getDate
 };
 
+var getDateTimeComposition = function (ds){
+    ds = ds.padEnd(17, '0');
+    var year = ds.substring(0,4);
+    var month = ds.substring(4,6);
+    var day = ds.substring(6,8);
+    var hours = ds.substring(8,10);
+    var minutes = ds.substring(10,12);
+    var seconds = ds.substring(12,14);
+    var milliseconds = ds.substring(14, 17);
+    var dateTimeComposition = {
+        'year': year,
+        'month': month,
+        'day': day,
+        'hours': hours,
+        'minutes': minutes,
+        'seconds': seconds,
+        'milliseconds': milliseconds
+    };
+    return dateTimeComposition;
+};
+
+// handling the datetime format here
+var getDateTime = function (dateTimeString) {
+    if (!validString(dateTimeString))
+        return '';
+
+    // handle special datetime format like 20130130080051+0500, 20130130080051-0500
+    var ds = dateTimeString.toString();
+    if (/^(\d{14})(-|\+)(\d{4})$/.test(ds))
+    {
+        var dateSections = ds.split(ds[14]);
+        var dateTimeComposition = getDateTimeComposition(dateSections[0]);
+        var date = dateTimeComposition.year + '-' + dateTimeComposition.month + '-' + dateTimeComposition.day;
+        var time = dateTimeComposition.hours + ':' + dateTimeComposition.minutes + ':' + dateTimeComposition.seconds;
+        var timezone = ds[14] + dateSections[1];
+        if (!validUTCDateTime(dateTimeComposition))
+            throw `Invalid datetime: ${ds}`;
+        return new Date(date + ' ' + time + ' ' + timezone).toISOString();
+    }
+
+    ds = validDigitString(dateTimeString);
+    if (ds.length <= 8)
+        return convertDate(ds);
+    
+    // Padding 0s to 17 digits 
+    dateTimeComposition = getDateTimeComposition(ds);
+    if (!validUTCDateTime(dateTimeComposition))
+        throw `Invalid datetime: ${ds}`;
+    return (new Date(Date.UTC(dateTimeComposition.year, dateTimeComposition.month - 1, dateTimeComposition.day, dateTimeComposition.hours, dateTimeComposition.minutes, dateTimeComposition.seconds, dateTimeComposition.milliseconds))).toJSON();
+};
+
+module.exports.internal = {
+    getDateTime: getDateTime
+};
+
```

Comment:
Does it override the declaration above for getDate?

Reply:
Yes, I modified it